test/libsolidity/syntaxTests/functionTypes/delete_function_type.sol

```diff
@@ -3,15 +3,13 @@ contract C {
     function(uint) internal returns (uint) y;
     function f() public {
         delete x;
-        var a = y;
+        function(uint) internal returns (uint) a = y;
         delete a;
         delete y;
-        var c = f;
+        function() internal c = f;
         delete c;
         function(uint) internal returns (uint) g;
         delete g;
```

Comment:
Not the job of this PR, but could you also add an external function type and delete it?
`function(uint) external returns (uint) r = this.f; delete r;`?